ext/page_log/palm/palm.c

```diff
@@ -532,6 +532,49 @@ palm_add_reference(WT_PAGE_LOG *page_log)
     return (0);
 }
 
+/*
+ * palm_abandon_checkpoint --
+ *     Abandon an incomplete checkpoint. This is an optional function, required only for internal
+ *     testing.
+ */
+static int
+palm_abandon_checkpoint(WT_PAGE_LOG *page_log, WT_SESSION *session, uint64_t last_checkpoint_lsn)
+{
+    PALM *palm;
+    PALM_KV_CONTEXT context;
+    uint64_t lsn;
+    int ret;
+
+    palm = (PALM *)page_log;
+    palm_init_context(palm, &context);
+
+    PALM_KV_RET(palm, session, palm_kv_begin_transaction(&context, palm->kv_env, false));
+    if (last_checkpoint_lsn == 0) {
+        ret = palm_kv_get_last_checkpoint(&context, &lsn, NULL, NULL, NULL);
+        if (ret == MDB_NOTFOUND)
+            last_checkpoint_lsn = 0;
```

Comment:
I'm concerned that `last_checkpoint_lsn = 0` ends up having two different meanings: it's used as a flag for "load the most recent checkpoint" but at the same time, `0` is also a legitimate value meaning "abandon all checkpoints" (and the code behaves that way if `NOTFOUND` is returned here).